### DISCUSSION TITLE
请问如何在header或Item模板内区分自动标签和手动标签

### DISCUSSION THREAD

**Discussion by C1ngCam** [windingwind/zotero-better-notes#1091] [Q&A] [answered]
在 Zotero 中，手动创建的标签值为 0，而从文献中提取的标签值为 1。
一些插件中是使用以下代码来实现的单独提取手动标签：
```
it.tags.filter(t => t.type === 0)
```
我尝试使用以下代码但好像并没有效果
```
yaml.tags = noteItem.parentItem.getTags()
  .filter((_t) => _t.type === 0)  
  .map((_t) => _t.tag);           
```

**Comment by windingwind** [comment]
我确认可正常工作，请截图完整模板编辑器界面

**Reply by C1ngCam** [reply]
谢谢大佬的回复，以下为我的header模板编辑器界面
![image](https://github.com/user-attachments/assets/b3f39cac-5af4-4e27-a3ee-0d19716c9be0)
可以看到Tags内容为空并且成了[]为外框，但是如果用您内置的tags语法则没有问题正常导出。
![image](https://github.com/user-attachments/assets/8d9ed8e2-909f-460c-8dc4-26d59fd940c7)
我是想通过type筛选得到类型为0的手动标签并放在yaml内，再次感谢您抽时间查看我的问题！

**Reply by windingwind** [reply]
可以检查一下是否这些标签都是分类为1的自动标签。由插件产生的标签，可能也会归属于自动标签

**Reply by C1ngCam** [reply]
抱歉再次打扰，我确定即便是手动添加的标签也依然是“[]"为空
如果可以的话能否麻烦看一下您能够实现的代码格式或者模板编辑器界面呢？

**Reply by windingwind** [reply, marked as answer]
用
```js
header.tags = noteItem.parentItem.getTags()
  .filter((_t) => !_t.type)
  .map((_t) => _t.tag);
```

**Reply by C1ngCam** [reply]
谢谢您指导
